packages/command/src/sfdxCommand.ts

```diff
@@ -238,10 +238,7 @@ export abstract class SfdxCommand extends Command {
       // Throw an error if the command requires a devhub and there is no targetdevhubusername
       // flag set and no defaultdevhubusername set.
       if (this.statics.requiresDevhubUsername) {
-        if (err.name === 'NoUsername') {
-          throw SfdxError.create('@salesforce/command', 'command', 'RequiresDevhubUsernameError');
-        }
-        throw err;
+        throw SfdxError.create('@salesforce/command', 'command', 'RequiresDevhubUsernameError');
```

Comment:
The previous code is odd (redundant). It makes me wonder if there's some other type of error that's no longer being picked up?

Reply:
The previous code seemed correct.I think you still need to add "throw err" at the end, incase there is some other error apart from devhub missing.

catch (err) {
      // Throw an error if the command requires a devhub and there is no targetdevhubusername
      // flag set and no defaultdevhubusername set.
      if (this.statics.requiresDevhubUsername) {
        throw SfdxError.create('@salesforce/command', 'command', 'RequiresDevhubUsernameError');
      }
throw err ; // in case its not due to missing devhub.
    }

Reply:
Ah, I didn't mean the code was wrong, it just seemed a little odd to me. There is an initial check of if (this.statics.requiresDevhubUsername), which seems to imply that you would only make it into that block if `requiresDevhubUsername` was set. Which would seem to make if (err.name === 'NoUsername') redundant. But, I would definitely want to ensure we're not missing any other potential errors that would make it through. 

Does that make an argument for adding more checks?

Reply:
take a look now 👍